### ISSUE TITLE
NoticePresenter & keyboard

### ISSUE THREAD

**Issue by Ecarrion** [woocommerce/woocommerce-ios#4942]
During the development of the Order Editing project, we noticed that notices presented by the `NoticePresenter` are hidden when the keyboard is present.

- Tasks

- [ ] Update `NoticePresenter` to move the notice up/down per keyboard presence
- [ ] Test changes don't break current notices presentation

**Comment by shiki** [comment]
For reference in case it's useful: This is how it's done in WordPress' NoticePresenter: 

https://github.com/wordpress-mobile/WordPress-iOS/blob/6e1384d54fe0988854b4a6405934154d21754bee/WordPress/Classes/ViewRelated/System/Notices/NoticePresenter.swift#L99-L134

**Comment by selanthiraiyan** [comment]
Hello 👋 
I would like to take a look into this issue.

**Comment by jaclync** [comment]
@Ecarrion Are there steps to reproduce this issue in the order editing flow? @selanthiraiyan's PR https://github.com/woocommerce/woocommerce-ios/pull/5120 is tested with product editing, maybe we can include the steps for order editing (original issue) as well. I tried editing a few fields of an order but didn't find where the keyboard can come up while a notice is presented 🤔

**Comment by shiki** [comment]
@jaclync IIRC, I was able to reproduce the problem when:

1. I was offline.
2. I added/saved a shipment tracking number. A notice with an error was shown but it was hidden behind a keyboard.

**Comment by Ecarrion** [comment]
Hi @jaclync, 

I filed this error for the following scenario:

- Go to edit an order note
- Turn on wifi
- Tap on the "Done" button.

Then the "Error notice" was hidden under the keyboard. Very similar to the situation @shiki mentions.

I did however prevent that error from happening by dismissing the keyboard before presenting a notice.
That's why you probably can't reproduce the error. 😅 

https://github.com/woocommerce/woocommerce-ios/blob/edc9909b31492656e42d66dfd11d943224a2a41e/WooCommerce/Classes/ViewRelated/Orders/Order%20Details/Order%20Notes%20Section/Customer%20Note/EditCustomerNote.swift#L39-L40
